sycl/test-e2e/Graph/event_profiling_info.cpp

```diff
@@ -178,6 +182,25 @@ int main() {
     assert(verifyProfiling(CopyEvent) && verifyProfiling(KernelEvent1) &&
            verifyProfiling(KernelEvent2) &&
            compareProfiling(KernelEvent1, KernelEvent2));
+
+    // Checks exception thrown if profiling is requested while in-order
+    // optimization enabled. Note that in-order cmd-list optmization is only
+    // available for level-zero backend.
+    auto Backend = Dev.get_backend();
+    if (Backend == backend::ext_oneapi_level_zero) {
```

Comment:
See spec comment that I think all backends, not just L0, should throw if the enable_profiling property isn't set and profiling is done.